File: src/main/scala/com/github/mauricio/postgresql/messages/frontend/PreparedStatementOpeningMessage.scala
package com.github.mauricio.postgresql.messages.frontend

import com.github.mauricio.postgresql.messages.backend.Message

/**
 * User: Maurício Linhares
 * Date: 3/12/12
 * Time: 1:00 AM
 */

class PreparedStatementOpeningMessage( query : String, values : Seq[Any] )
  extends PreparedStatementMessage(Message.Parse, query, values)

File: postgresql-async/src/main/java/com/github/mauricio/async/db/postgresql/messages/frontend/PreparedStatementOpeningMessage.kt
package com.github.mauricio.async.db.postgresql.messages.frontend

import com.github.jasync.sql.db.column.ColumnEncoderRegistry
import com.github.mauricio.async.db.postgresql.messages.backend.ServerMessage

class PreparedStatementOpeningMessage(statementId: Int, query: String, values: List<Any?>, encoderRegistry: ColumnEncoderRegistry)
  : PreparedStatementMessage(statementId, ServerMessage.Parse, query, values, encoderRegistry) {

  override fun toString(): String = "${this.javaClass.simpleName}(id=$statementId,query=$query,values=$values})"
}
